### PR TITLE
Add author id and timestamp of modification to annotations

### PR DESCRIPTION
**Pull request by DE-AI**
Augmented Django models for annotations for storing author id and timestamp in SQL database. Changes solely affect the database.

### Motivation and context

In cases of multiple contributors/annotators, it is relevant to keep track of the actual authors of annotations and the date and time of modification. Being aware that jobs can be assigned to annotators, it is nevertheless relevant if annotations have been changed thereafter by task/project owners. The timestamp may be relevant for estimating approximately the time required in an annotation process by certain annotators and for obtaining an idea for future annotation tasks.
The pull request adds the author-id and a timestamp to SQL tables containing annotations (i.e., engine_labeledimage, engine_labeledtrack, engine_labeledshape). Associated migration scripts have been added which fill added fields of previous annotations with NULL.

### How has this been tested?

All changes have been tested with an existing database executing 'python3 manage.py makemigrations / migrate' and checking the modifications in the SQL database. New annotations successfully store the author id and timestamp of modification in the database. Both tests yield a successful evaluation.

### Checklist

- [x] I submit my changes into the `develop` branch
- ~~[ ] I have added description of my changes into [CHANGELOG]( engine_) file~~
- ~~[ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly~~
- ~~[ ] I have added tests to cover my changes~~
- ~~[ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))~~
- ~~[ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))~~

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- ~~[ ] I have updated the license header for each file (see an example below)~~

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by nmanovic** [comment]
@azhavoro , could you please look at the PR?

File: cvat/apps/engine/migrations/0042_author_annotation.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Generated by Django 3.1.10 on 2021-06-16 09:00
```
Comment (azhavoro) [inline cvat/apps/engine/migrations/0042_author_annotation.py:1]: Please combine the migration files into one

**Comment by azhavoro** [comment]
@DE-AI Hi, This is not work in my case:
1. deploy cvat from your branch
2. create task and create any shape
3. save
4. reload task
5. I got `author_id: null` for new annotations: 
```json
{
  "type": "rectangle",
  "occluded": false,
  "z_order": 0,
  "points": [
    485.140625,
    88.078125,
    794.0823059082031,
    224.92477416992188
  ],
  "id": 2,
  "frame": 0,
  "label_id": 4,
  "author_id": null,
  "group": 0,
  "source": "manual",
  "attributes": []
}
```

Also, please resolve conflicts.

**Comment by lixinkuicomputer** [comment]
Hello，could you provide docker-image or docker-deployment-steps？ Thanks a lot。

**Comment by azhavoro** [comment]
> Hello，could you provide docker-image or docker-deployment-steps？ Thanks a lot。

Hi, if the question is addressed to me, the reprodusion steps with docker are pretty simple:

1. git clone git@github.com:DE-AI/cvat.git cvat_deai --branch annotation-attributes-db-extension
2. cd cvat_deai
3.  docker-compose -f docker-compose.yml -f docker-compose.dev.yml up -d --build
4. create superuser and login
5. create a task
6. add any shape and save
7. reload task
8. `author_id` field is null for the added shape

**Comment by nmanovic** [comment]
@DE-AI , don't hesitate to reopen the PR if you have time to update it. Thanks for the contribution but it has a number of problems which should be resolved before being merged. To keep the number of open issues and PRs manageable I will close the PR now.

**Comment by hamzahzmalik** [comment]
@nmanovic 
Hi,

I was exploring the CVAT db and came across the author_id field in the engine_jobcommit table. However, the field is empty for all annotations ever made. 

Are annotations tracked by CVAT with regard to their author and date/time? 

This is indeed a needed and useful feature, being able to track and trace the annotations back to their author as well as the time the changes were made. If CVAT already documents this information somewhere can you please share more details on that?

Thanks.

**Comment by nmanovic** [comment]
@hamzahzmalik , we did some steps in the direction but they were not finished yet.

**Pull request closed without merging**